snap/hooks/install

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -e
+
+# get the values of $SNAP_DATA and $SNAP using the current symlink instead of
+# the default behavior which has the revision hard-coded, which breaks after
+# a refresh
+SNAP_DATA_CURRENT=${SNAP_DATA/%$SNAP_REVISION/current}
+SNAP_CURRENT=${SNAP/%$SNAP_REVISION/current}
+
+# install all the config files from $SNAP/config/SERVICE/res/configuration.toml
+# into $SNAP_DATA/config and replace $SNAP_COMMON environment variable in the config files
+mkdir -p "$SNAP_DATA/config"
+if [ ! -f "$SNAP_DATA/config/app-service-configurable/res/configuration.toml" ]; then
+    mkdir -p "$SNAP_DATA/config/res"
+    mkdir -p "$SNAP_DATA/config/res/http-export"
+    mkdir -p "$SNAP_DATA/config/res/mqtt-export"
+    mkdir -p "$SNAP_DATA/config/res/rules-engine"
+    mkdir -p "$SNAP_DATA/config/res/push-to-core"
+
+    for file in configuration.toml "http-export/configuration.toml" "mqtt-export/configuration.toml" "rules-engine/configuration.toml" "push-to-core/configuration.toml"; do
+        cp "$SNAP/config/res/$file" "$SNAP_DATA/config/res/$file"
+
+        sed -i -e "s@\$SNAP_COMMON@$SNAP_COMMON@g" \
```

Comment:
you may only need to replace `$SNAP_COMMON` right now, but it's conceivable you (or someone else) might need to replace other vars in the configuration.toml file and might be confused why the env vars don't get substituted when $SNAP_COMMON does. Doing all the env vars like `$SNAP_DATA`, `$SNAP`, and `$SNAP_COMMON` I think is what we want to do